beep/structure.py

```diff
@@ -304,19 +306,26 @@ def get_interpolated_cycles(
             reg_cycles = [i for i in self.data.cycle_index.unique()]
 
         v_range = v_range or [2.8, 3.5]
+
+        # If any regular cycle contains a waveform step, interpolate on test_time.
+        if self.data[self.data.cycle_index.isin(reg_cycles)].\
+                groupby(["cycle_index", "step_index"]).\
+                apply(determine_whether_step_is_waveform).any():
+            discharge_axis = 'test_time'
```

Comment:
@patrickherring-TRI this moves the logic for waveform detection one level above, right after splitting out regular and diagnostic cycles.